App.js

```diff
@@ -1,9 +1,363 @@
 import React, { useState, useCallback } from 'react';
-import { StyleSheet, Text, View, FlatList, Dimensions, TouchableOpacity} from 'react-native';
+import { StyleSheet, Text, View, FlatList, Dimensions, TouchableHighlight } from 'react-native';
 import { newBoard } from './newboard.js';
 
 export default function App() {
-  
+  const [{ roll, even, odd, position, message, optMessage, score, board }, setGameState] = useState({
+    roll: 0,
+    even: false,
+    odd: false,
+    position: 0,
+    message: 'Roll again',
+    optMessage: 'Kim',
+    score: 0,
+    board: JSON.parse(JSON.stringify(newBoard)),
+  });
+
+  // Nav bar
+  const nav = [
+    { key: 1, name: 'PLAY', invisible: false },
+    { key: 2, name: '', invisible: true },
+    { key: 3, name: '', invisible: true },
+    { key: 4, name: '', invisible: true },
+    { key: 5, name: '', invisible: true },
+    { key: 6, name: 'Score' },
+    { key: 7, name: 0 },
+  ];
+
+  const numColumns = 7;
+
+  // render Nav Bar
+  const renderNav = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    if (item.name === 'PLAY') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressPlay()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    if (item.key === 7) {
+      return (
+        <View style={styles.item}>
+          <Text style={styles.itemText}>{score}</Text>
+        </View>
+      );
+    }
+    return (
+      <View style={styles.item}>
+        <Text style={styles.itemText}>{item.name}</Text>
+      </View>
+    );
+  };
+
+  // render board
+  const renderBoard = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    // Show ROLL button
+    if (item.name === 'ROLL') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressRoll(item.key)}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Even button and mark it if it has been selected
+    if (item.name === 'EVEN') {
+      return (
+        <View style={even ? [styles.item, styles.markSpot] : styles.item}>
+          <TouchableHighlight onPress={() => pressEven()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Odd button and mark it if it has been selected
+    if (item.name === 'ODD') {
+      return (
+        <View style={odd ? [styles.item, styles.markSpot] : styles.item}>
```

Comment:
here is another example of where you can do some code sharing between the EVEN and ODD cases.
Here is some example code:

```
else if (item.name === 'EVEN' || item.name === 'ODD') {
  const onPress = item.name === 'EVEN' ? pressEven : onPressOdd;
  return (
    <View style={even ? [styles.item, styles.markSpot] : styles.item}>
      <TouchableHighlight onPress={onPress}>
      <Text style={styles.itemText}>{item.name}</Text>
      </TouchableHighlight>
    </View>
  );
}
```